### PR TITLE
apple-macbookpro-14-1: update, apple-imac-18-2: add

### PR DESCRIPTION
**Pull request by paepckehh**
###### Description of changes

apple-macbookpro-14-1:
* update README.md (including links to pending PRs)
* update default.nix, simplify, follow upstream, fix version typo

apple-imac-18-2:
* introduce imac line
* introduce imac-18-2 model 

(merged PR for similar producs/requests updates to save reviewers resources)

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: apple/imac/18-2/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ [x] ok
+
+## NVME
+ [x] ok, older NVME / Controller may need workaround for resume
+
+## Suspend/Resumer
+ [ ] Thunderbolt, WIFI, NVME may still need reboot (sometimes).
+
+## Wifi
+ [x] Working (2,4Ghz & 5Ghz supported), WEP3 broken -> brcm fw blob (2015)
```
Comment (Mic92) [inline apple/imac/18-2/README.md:22]:
```suggestion
- [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322968

## Bluetooth
- [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322964

## Thunderbolt
- [x] ok

## SATA
- [x] ok

## NVME
- [x] ok, older NVME / Controller may need workaround for resume

## Suspend/Resumer
- [ ] Thunderbolt, WIFI, NVME may still need reboot (sometimes).

## Wifi
- [x] Working (2,4Ghz & 5Ghz supported), WEP3 broken -> brcm fw blob (2015) 
```

**Comment by paepckehh** [comment]
Is this change supposed to land before or after the common reorg?

**Comment by Mic92** [comment]
before.

File: apple/imac/18-2/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# iMac 18,2, NixOS 24.05 (2024)
+
+## Audio
+- [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322968
+
+## Bluetooth
+- [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322964
```
Comment (Mic92) [inline apple/imac/18-2/README.md:7]:
```suggestion
- [ ] broken until https://github.com/NixOS/nixpkgs/pull/322968 lands in master

## Bluetooth
- [ ] broken until https://github.com/NixOS/nixpkgs/pull/322964 lands in master
```

File: apple/macbook-pro/14-1/README.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
-# MacBook Pro 14,1, NixOS 24.01 (2024)
+# MacBook Pro 14,1, NixOS 24.05 (2024)
 
 ## Audio
- [ ] Still broken, use usb/hdmi instead, nixos pkg needed: https://github.com/davidjo/snd_hda_macbookpro
+ - [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322968
```
Comment (Mic92) [inline apple/macbook-pro/14-1/README.md:4]:
```suggestion
 - [ ] broken until https://github.com/NixOS/nixpkgs/pull/322968 lands in master
```

File: apple/macbook-pro/14-1/README.md [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
 
 ## Bluetooth
- [ ] Still broken, even (hacky) workaround does not work any more with latest driver updates
+ - [ ] broken, PR pending: https://github.com/NixOS/nixpkgs/pull/322964
```
Comment (Mic92) [inline apple/macbook-pro/14-1/README.md:7]:
```suggestion
 - [ ] broken lands https://github.com/NixOS/nixpkgs/pull/322964 in master
```

**Comment by Mic92** [comment]
Was this conflict resolution correct?

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
@mergify requeue

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
@mergify rebase

**Pull request merged**